### PR TITLE
added QWen3 EAGLE model

### PR DESCRIPTION
**Pull request by w32zhong**
## Motivation
SImilar PR: https://github.com/sgl-project/sglang/pull/2863

Although eagle-3 generally achieves higher speedups, eagle-2 requires far less training time. Adding eagle-2 to recent models could remain beneficial.

For example:
* When accelerating customized small models in resource-constrained environments.
  For eagle (v1/v2) to train a 7B model, a single RTX 3090 with 24GiB of CUDA memory is sufficient (ref: https://arxiv.org/pdf/2401.15077), and 8x less training data is used compared to eagle-3 official checkpoints.
* And in research, this can be used as a baseline option to compare speculative algorithms in batched serving scenarios.

## Modifications
A single dense qwen3 version for now, but I am open to further adding eagle for qwen-moe if you want.

## Accuracy Tests

On a RTX 5070 Ti (sm120a)
```sh
python -m sglang.launch_server \
  --model Qwen/Qwen3-4B-Instruct-2507 \
  --speculative-algo EAGLE  \
  --speculative-draft-model-path w32zhong/sglang-EAGLE-Qwen3-4B-Instruct-2507 \
  --speculative-num-steps 6 \
  --speculative-eagle-topk 10 \
  --speculative-num-draft-tokens 60 \
  --mem-fraction 0.7 \
  --disable-cuda-graph # for faster test
```

```sh
python bench_sglang_eagle.py --parallel 1 --num-questions 80
```
```
#questions: 80, Throughput: 87.15 token/s, Acceptance length: 3.08
```
I don't see an official eagle(v1/v2) draft model for Qwen3, but the accept length should be valid since this checkpoint was trained using a prototype framework that has been verified to align with the official EAGLE + LLaMA-2 accept length. (I’m ok to transfer [this test checkpoint](https://huggingface.co/w32zhong/sglang-EAGLE-Qwen3-4B-Instruct-2507) to lmsys if needed)

On a RTX 4070 TiS (sm89), here are some more comprehensive results with CUDA graph **on/off**:

| accept length | bs=1, tree=6,10,60 | bs=4, tree=6,10,60 |
|:-------------------------:|:----------------:|:----------------:|
| 3.07              | 98.79/90.74      | 236.35/225.97    |

| accept length | bs=1, tree=3,1,4 | bs=4, tree=3,1,4 | bs=8, tree=3,1,4 |
|:------------------:|:--------------:|:--------------:|:--------------:|
| 1.96               | 90.04/73.14    | 316.89/272.38  | 504.65/476.0   |

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

File: python/sglang/srt/models/qwen3_eagle.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                hidden_states = self.embed_tokens(input_ids)
+            else:
+                hidden_states = input_embeds
+
+            prev_states = forward_batch.spec_info.hidden_states
+            hidden_states = self.eagle_fc(
+                torch.cat((hidden_states, prev_states), dim=-1)
+            )
+        else:
+            assert "pp_proxy_tensors" in kwargs
```
[the remark this answers is not in the document]
Comment (w32zhong) [reply python/sglang/srt/models/qwen3_eagle.py:99]: this will be handled in origin_forward.
